packages/core/http/core-http-router-server-internal/src/router.ts

```diff
@@ -271,3 +272,25 @@ type WithoutHeadArgument<T> = T extends (first: any, ...rest: infer Params) => i
 type RequestHandlerEnhanced<P, Q, B, Method extends RouteMethod> = WithoutHeadArgument<
   RequestHandler<P, Q, B, RequestHandlerContextBase, Method>
 >;
+
+const FORBIDDEN_HEADERS = [
+  'authorization',
+  'cookie',
+  'set-cookie',
+  'x-elastic-app-auth',
+  'es-client-authentication',
+];
+
+const REDACTED_HEADER_TEXT = '[REDACTED]';
+
+const redactSensitiveHeaders = (data: any) => {
+  // keep the original Error object, but deep clone all its properties
+  // this allows redacting some fields without modifying the original objects,
+  // and at the same time preserving the original Error object
+  Object.entries(data).forEach(([key, value]) => {
+    data[key] = cloneDeepWith(value, (v, k) =>
```

Comment:
We're not doing anything else with the `error` object after that, so we should be fine.
I did not want to create a `new Error()` to avoid making the chain of causes more complex.
And if I cloned the whole `Error` into a plain object (and redacted that one instead) I was losing the `toString()` formatting, so I thought this trade off would be acceptable. Any ideas to improve this?

Reply:
UPDATE: This seems to impact one of the integration tests, so it looks like the `hapiResponseAdapter` keeps a reference to the error. I will try to clone it instead.